test/bundler/bundler_html.test.ts

```diff
@@ -0,0 +1,470 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <link rel="stylesheet" href="../styles/page2.css">
+    <script src="../scripts/page2.js"></script>
+  </head>
+  <body>
+    <h1>Page 2</h1>
+  </body>
+</html>`,
+      "/in/styles/page1.css": `
+@import './shared.css';
+.page1 { font-size: 20px; }`,
+      "/in/styles/page2.css": `
+@import './shared.css';
+.page2 { font-size: 18px; }`,
+      "/in/styles/shared.css": `
+@import './reset.css';
+.shared { color: blue; }`,
+      "/in/styles/reset.css": `
+* { box-sizing: border-box; }`,
+      "/in/scripts/page1.js": `
+import { sharedUtil } from './shared.js';
+import { largeModule } from './large-module.js';
+console.log('Page 1');
+sharedUtil();`,
+      "/in/scripts/page2.js": `
+import { sharedUtil } from './shared.js';
+import { largeModule } from './large-module.js';
+console.log('Page 2');
+sharedUtil();`,
+      "/in/scripts/shared.js": `
+export const sharedUtil = () => console.log('Shared utility');`,
+      "/in/scripts/large-module.js": `
+export const largeModule = {
+  // Simulate a large shared module
+  bigData: new Array(1000).fill('data'),
+  methods: { /* ... */ }
+};`,
+    },
+    entryPoints: ["/in/pages/page1.html", "/in/pages/page2.html"],
+    experimentalHtml: true,
+    experimentalCss: true,
+    splitting: true,
+    onAfterBundle(api) {
+      // Check both pages
+      for (const page of ["page1", "page2"]) {
+        api.expectFile(`out/${page}.html`).toMatch(/href=".*\.css"/);
+        api.expectFile(`out/${page}.html`).toMatch(/src=".*\.js"/);
+        api.expectFile(`out/${page}.html`).not.toContain(`${page}.css`);
+        api.expectFile(`out/${page}.html`).not.toContain(`${page}.js`);
+      }
+
+      // Verify that shared code exists in both bundles
+      const page1Html = api.readFile("out/page1.html");
+      const page2Html = api.readFile("out/page2.html");
+
+      const page1JsPath = page1Html.match(/src="(.*\.js)"/)?.[1];
+      const page2JsPath = page2Html.match(/src="(.*\.js)"/)?.[1];
+
+      expect(page1JsPath).toBeDefined();
+      expect(page2JsPath).toBeDefined();
+
+      const page1Js = api.readFile("out/" + page1JsPath!);
+      const page2Js = api.readFile("out/" + page2JsPath!);
+
+      // Check we imported the shared module
+      expect(page2Js).toContain("import{sharedUtil}");
+      expect(page1Js).toContain("import{sharedUtil}");
+
+      // Check CSS bundles
+      const page1CssPath = page1Html.match(/href="(.*\.css)"/)?.[1];
+      const page2CssPath = page2Html.match(/href="(.*\.css)"/)?.[1];
+
+      expect(page1CssPath).toBeDefined();
+      expect(page2CssPath).toBeDefined();
+
+      const page1Css = api.readFile("out/" + page1CssPath!);
+      const page2Css = api.readFile("out/" + page2CssPath!);
+      expect(page1Css).toContain("box-sizing:border-box");
+      expect(page2Css).toContain("box-sizing:border-box");
+      expect(page1Css).toContain(".shared");
+      expect(page2Css).toContain(".shared");
+    },
+  });
```

Comment:
can a test that has a js entrypoint importing an html file be added

additionally a js file importing an html file importing the first js file

i assume these already work but would be good to have it tested